### PR TITLE
Bug/ccp 226

### PR DESCRIPTION
**Pull request by kkrish39**

### CODE REVIEW

**Review by nowke — APPROVED** [review]

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
a couple things:
- "Search on description" should be "Search in description field" and there should be no period afterwards
- when "Search in description field" is checked, word **and** description fields should be searched, not just description
- the pagination doesn't seem to pass the search in description flag

**Comment by jdamerow** [comment]
Resolve conflicts please

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/core/impl/ConceptManager.java [hunk 1]
```diff
@@ -146,14 +146,17 @@ public ConceptEntry getWordnetConceptEntry(String wordnetId) throws LuceneExcept
...
                     throws LuceneException, IllegalAccessException, IndexerRunningException {
         if (pos == null)
             return null;
 
-        Map<String, String> fieldMap = new HashMap<String, String>();
+        Map<String, String> fieldMap = new HashMap<>();
+        
         fieldMap.put(SearchFieldNames.WORD, word);
+        if(isSearchOnDescription)
+            fieldMap.put(SearchFieldNames.DESCRIPTION, word );
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/core/impl/ConceptManager.java:159]: please follow our code style

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
something is wrong now (I think it's the use of the TermQuery). If I search for Einstein and the description checkbox is checked, it only finds one result (probably the one that has einstein in the description) and none of the others.

**Comment by kkrish39** [comment]
Can you check once? I searched for einstein with the checkbox checked. I got three rows.
Screen Shot 2020-09-16 at 2 17 15 PM

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
It's still only returned one result when searching on description. Did you reindex after you did the changes to the query?

**Comment by kkrish39** [comment]
> It's still only returned one result when searching on description. Did you reindex after you did the changes to the query?

Let me check once again!

**Comment by kkrish39** [comment]
> It's still only returned one result when searching on description. Did you reindex after you did the changes to the query?

I reindexed and tried once again. Without checkbox checked.
Screen Shot 2020-10-27 at 11 58 27 AM

With checkbox checked.
Screen Shot 2020-10-27 at 11 58 41 AM

**Comment by jdamerow** [comment]
Exactly, the second search should not return fewer results but more. Why does it not find "Einstein's special theory of relativity"?

**Comment by kkrish39** [comment]
> Exactly, the second search should not return fewer results but more. Why does it not find "Einstein's special theory of relativity"?

I see what's happening and what is required. When you said the query should search "concept and description" I thought the combined search of both. 

Now the query is returning only the rows which contain the search term in both concept and description.

**Comment by jdamerow** [comment]
ah, yes, it should be "OR" not "AND"

**Comment by kkrish39** [comment]
I'm pretty sure it can be handled a bit more graciously. Let me know your comments.

**Review by ashmichheda — APPROVED** [review]

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/lucene/impl/LuceneUtility.java [hunk 2]
```diff
@@ -533,7 +535,7 @@ public void indexConcepts(String userName)
                 String searchString = fieldMap.get(search.fieldName());
                 if (searchString != null) {
                     searchString = searchString.toLowerCase();
-                    buildQuery(occur, perFieldAnalyzerWrapper, qBuild, builder, luceneFieldAnnotation, searchString);
+                    buildQuery(occur, perFieldAnalyzerWrapper, qBuild, builder, luceneFieldAnnotation, searchString, isSearchOnDescription);
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/lucene/impl/LuceneUtility.java:538]: isn't the only thing you need to do, test if `isSearchOnDescription` is true and if search is the description field, then exclude or not exclude it here?

Comment (jdamerow) [reply Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/lucene/impl/LuceneUtility.java:538]: something like 
```
if (search.fieldName != SearchFieldNames.DESCRIPTION || isSearchOnDescription ) { // or however you get the filed name
 buildQuery(...)
}
```

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Pull request closed without merging**